plugins/nf-amazon/src/main/nextflow/cloud/aws/nio/S3Path.java

```diff
@@ -410,12 +411,12 @@ public URI toUri() {
 		builder.append("s3://");
 		if (fileSystem.getEndpoint() != null) {
 			builder.append(fileSystem.getEndpoint());
-		}
-		builder.append("/");
+            builder.append("/");
+        }
 		builder.append(bucket);
```

Comment:
This may have unintended side effects. If I remember correctly, the syntax `s3:///foo/some-file.txt` should identity the lack of the host name in the URI (and therefore use the default one). Instead, when a custom endpoint is specified it should be `s3://some.host/foo/some-file.txt`.

This is why the `builder.append("/")` was outside the `if` branch